### ISSUE TITLE
Installation Error:  Could not find a version that satisfies the requirement

### ISSUE THREAD

**Issue by renjiege** [IntelPython/mkl_fft#26]
System: OS High Sierra 10.13.4 
Python: 3.7
Pip: 18.0

sudo -H pip3 install mkl-ffl

Error message:

Collecting mkl-ffl
 Could not find a version that satisfies the requirement mkl-ffl (from versions: )
No matching distribution found for mkl-ffl

 the same error message appears when I tried to install mkl-random

**Comment by oleksandr-pavlyk** [comment]
Yes, only 2.7, 3.5, and 3.6 wheels are currently available it seems.

**Comment by renjiege** [comment]
Thanks @oleksandr-pavlyk. I just created a virtual environment with python 3.5 as the interpreter. All the other libraries can be installed normally in the venv except for mkl-fft and mkl-random.

**Comment by kebaek** [comment]
This can be solved by running conda install -c intel mkl_fft instead of pip.

**Comment by Keerthana2701** [comment]
ERROR: Could not find a version that satisfies the requirement mkl-service==2.3.0 (from -r /tmp/build_92d7e5cf_/requirements.txt (line 36)) (from versions: none)
       ERROR: No matching distribution found for mkl-service==2.3.0 (from -r /tmp/build_92d7e5cf_/requirements.txt (line 36))

please help

**Comment by debasissil-python** [comment]
While pushing files to `git push heroku master` , I'm getting the same issue -

```
remote:        ERROR: Could not find a version that satisfies the requirement mkl-random==1.2.1 (from -r /tmp/build_cb7d0db0/requirements.txt (line 37)) (from versions: 1.2.2)
remote:        ERROR: No matching distribution found for mkl-random==1.2.1 (from -r /tmp/build_cb7d0db0/requirements.txt (line 37))
remote:  !     Push rejected, failed to compile Python app.
```

Please help how to go by..

**Comment by oleksandr-pavlyk** [comment]
> While pushing files to `git push heroku master` , I'm getting the same issue -
> 
> 
> 
> ```
> 
> remote:        ERROR: Could not find a version that satisfies the requirement mkl-random==1.2.1 (from -r /tmp/build_cb7d0db0/requirements.txt (line 37)) (from versions: 1.2.2)
> 
> remote:        ERROR: No matching distribution found for mkl-random==1.2.1 (from -r /tmp/build_cb7d0db0/requirements.txt (line 37))
> 
> remote:  !     Push rejected, failed to compile Python app.
> 
> ```
> 
> 
> 
> Please help how to go by..

I am unable to extend any assistance if I can not reproduce the problem. "heroku" is a name for a remote in your git config, and I have no way of knowing what that it. 

Pushing does not trigger any building. This must be done in your precommit script. What is it running?

**Comment by pujaburman30** [comment]
ERROR: Could not find a version that satisfies the requirement intel-opencl-rt==2021.3.0 (from dpcpp_cpp_rt->mkl_fft) (from versions: none)
ERROR: No matching distribution found for intel-opencl-rt==2021.3.0 (from dpcpp_cpp_rt->mkl_fft)

Getting above error while running below to install mkl_fft in Google colab (Python : 3.7).
`!pip install mkl_fft`

Please help with a solution.

**Comment by Vyacheslav-Smirnov** [comment]
Hello,
The issue occurs due to missing of intel-opencl-rt==2021.3.0 for Linux which is required for dpcpp_cpp_rt 2021.3
Pip tries to install latest Perflibs thus issue appears
Till intel-opencl-rt==2021.3.0 is published I can suggest following workaround:
`python -m pip install --index-url https://pypi.anaconda.org/intel/simple --extra-index-url https://pypi.org/simple mkl_fft numpy==1.20.3 dpcpp_cpp_rt==2021.2`

Here "https://pypi.anaconda.org/intel/simple" is used to install intel numpy
More simple command (without intel numpy using or if you have numpy preinstalled) is:
`python -m pip install mkl_fft dpcpp_cpp_rt==2021.2`

**Comment by thaizasilva** [comment]
I had the same problem and removing mkl from requirements.txt solved it

**Comment by vtavana** [comment]
Closing this issue as this was inactive for a long time. 

The most recent version `mkl-fft-1.3.14` is available for python 3.9, 3.10, 3.11, 312. If you have any problems with the newest version please open a new issue.
